src/java.desktop/share/classes/sun/swing/FilePane.java

```diff
@@ -1753,6 +1753,20 @@ private void doFileSelectionModeChanged(PropertyChangeEvent e) {
     private void doMultiSelectionChanged(PropertyChangeEvent e) {
         if (getFileChooser().isMultiSelectionEnabled()) {
             listSelectionModel.setSelectionMode(ListSelectionModel.MULTIPLE_INTERVAL_SELECTION);
+
+            //Check if any files are selected before setting multi-Selection.
+            //If selected, retain them and update the selected files in FileChooser class.
+            File[] selectedFiles = null;
+            if (getFileChooser().getSelectedFiles().length != 0) {
+                selectedFiles = getFileChooser().getSelectedFiles();
+            } else if (getFileChooser().getSelectedFile() != null) {
+                selectedFiles = new File[1];
+                selectedFiles[0] = getFileChooser().getSelectedFile();
+            }
+
+            if (selectedFiles != null) {
+                getFileChooser().setSelectedFiles(selectedFiles);
+            }
         } else {
             listSelectionModel.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
             clearSelection();
```

Comment:
When we move from multi-selection enabled to disable. We are clearing the selection and calling setSelectedFiles(null). Do we need to replicate similar state update when we move from multi-selection disable to enable (clearing selection and calling setSelectedFile(null))?

It looks like somewhere we are not clearing SelectedFile() and it is causing this issue.

Reply:
If we replicate the same then previously selected files will be cleared and user wont be able to retain the selected files.

Reply:
Clearing the selection from multi to single is fine, but the reverse might not be right I guess..... Since selected file/files can be retained when user switch to multi selection.

Reply:
In this scenario we are actually closing the first JFileChooser dialog and then opening another JFileChooser dialog with multi-selection. Do we need to maintain selected file state between these dialogs? If yes, why? Please clarify.

Reply:
In keyboard selection and In mac, the selection is maintained during this state transition, only in this scenario its not handled. And since same filechooser is used in selecting the files ( Just pop-up dialogue is used to select the files) I think it would be better to retain the selection.

Reply:
In keyboard selection the state is actually not maintained. Then do you think it better to clear the selection and set selected files to null....?

Reply:
Clearing the selection and setting it to null also solves the problem though. When we call clear selection, reset will happen and selectedFiles will be set.

Reply:
As discussed on call its better to clear selection when we move between multi-selection enabled/disabled as captured in https://github.com/openjdk/jdk/pull/9996#discussion_r965791205

Reply:
Yeah, I have updated as discussed on call and tested.